tests/integration/test_recipe.py

```diff
@@ -341,64 +400,46 @@ def test_default_preprocessor(tmp_path, patched_datafinder, config_user):
 
     fix_dir = os.path.join(
         preproc_dir, 'CMIP5_CanESM2_Oyr_historical_r1i1p1_chl_2000-2005_fixed')
-    defaults = {
-        'load': {
-            'callback': concatenate_callback,
-        },
-        'concatenate': {},
-        'fix_file': {
-            'project': 'CMIP5',
-            'dataset': 'CanESM2',
-            'short_name': 'chl',
-            'mip': 'Oyr',
-            'output_dir': fix_dir,
-        },
-        'fix_data': {
-            'check_level': CheckLevels.DEFAULT,
-            'project': 'CMIP5',
-            'dataset': 'CanESM2',
-            'short_name': 'chl',
-            'mip': 'Oyr',
-            'frequency': 'yr',
-        },
-        'fix_metadata': {
-            'check_level': CheckLevels.DEFAULT,
-            'project': 'CMIP5',
-            'dataset': 'CanESM2',
-            'short_name': 'chl',
-            'mip': 'Oyr',
-            'frequency': 'yr',
-        },
-        'extract_time': {
-            'start_year': 2000,
-            'end_year': 2006,
-            'start_month': 1,
-            'end_month': 1,
-            'start_day': 1,
-            'end_day': 1,
-        },
-        'cmor_check_metadata': {
-            'check_level': CheckLevels.DEFAULT,
-            'cmor_table': 'CMIP5',
-            'mip': 'Oyr',
-            'short_name': 'chl',
-            'frequency': 'yr',
-        },
-        'cmor_check_data': {
-            'check_level': CheckLevels.DEFAULT,
-            'cmor_table': 'CMIP5',
-            'mip': 'Oyr',
-            'short_name': 'chl',
-            'frequency': 'yr',
-        },
-        'cleanup': {
-            'remove': [fix_dir]
-        },
-        'save': {
-            'compress': False,
-            'filename': product.filename,
-        }
-    }
+    defaults = _get_default_settings_for_chl(fix_dir, product.filename)
+    assert product.settings == defaults
+
+
+def test_default_preprocessor_custom_order(tmp_path, patched_datafinder,
```

Comment:
I'm not sure if I understand what is being tested here. Could you add a short docstring?

Reply:
Added docstring.